File: Examples/android/CMeSdkJava/app/src/main/java/me/connect/sdk/java/sample/connections/ConnectionsAdapter.java
package me.connect.sdk.java.sample.connections;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import org.w3c.dom.Text;

import java.util.ArrayList;
import java.util.List;

import me.connect.sdk.java.sample.R;
import me.connect.sdk.java.sample.db.entity.Connection;

public class ConnectionsAdapter extends RecyclerView.Adapter<ConnectionsAdapter.ConnectionsViewHolder> {

    private List<Connection> data;

    public ConnectionsAdapter() {
        data = new ArrayList<>();
    }

    @NonNull
    @Override
    public ConnectionsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.connection_list_item, parent, false);
        return new ConnectionsViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ConnectionsViewHolder holder, int position) {
        Connection conn = data.get(position);
        holder.text.setText(conn.name);
        if (conn.icon != null) {
            Glide.with(holder.image.getContext()).load(conn.icon).into(holder.image);
        }
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public void setData(List<Connection> data) {
        this.data = data;
        this.notifyDataSetChanged();
    }

    static class ConnectionsViewHolder extends RecyclerView.ViewHolder {
        public TextView text;
        public ImageView image;

        public ConnectionsViewHolder(View v) {
            super(v);
            text = v.findViewById(R.id.text);
            image = v.findViewById(R.id.image);
        }
    }
}

File: Examples/android/CMeSdkJava/app_kt/src/main/kotlin/me/connect/sdk/java/samplekt/connections/ConnectionsAdapter.kt
package me.connect.sdk.java.samplekt.connections

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import me.connect.sdk.java.samplekt.db.entity.Connection
import me.connect.sdk.java.samplekt.R

class ConnectionsAdapter : RecyclerView.Adapter<ConnectionsAdapter.ConnectionsViewHolder>() {
    private var data = mutableListOf<Connection>()
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ConnectionsViewHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.connection_list_item, parent, false)
        return ConnectionsViewHolder(v)
    }

    override fun onBindViewHolder(holder: ConnectionsViewHolder, position: Int) {
        val conn = data[position]
        holder.text.text = conn.name
        if (conn.icon != null) {
            Glide.with(holder.image.context).load(conn.icon).into(holder.image)
        }
    }

    override fun getItemCount(): Int = data.size

    fun setData(connections: List<Connection>) {
        data.clear()
        data.addAll(connections)
        notifyDataSetChanged()
    }

    class ConnectionsViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        var text: TextView = v.findViewById(R.id.text)
        var image: ImageView = v.findViewById(R.id.image)
    }

}